hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -332,10 +347,44 @@ def from_typing_type(thing):
             # Reversible is somehow a subclass of Hashable, so we tuplize it.
             # See also the discussion at https://bugs.python.org/issue39046
             typing.Reversible: st.lists(st.integers()).map(tuple),  # type: ignore
-            typing.SupportsAbs: st.complex_numbers(),
-            typing.SupportsComplex: st.complex_numbers(),
-            typing.SupportsFloat: st.floats(),
-            typing.SupportsInt: st.floats(),
+            typing.SupportsAbs: (
+                st.booleans()
+                | st.integers()
+                | st.floats()
+                | st.complex_numbers()
+                | st.fractions()
+                | st.decimals()
+                | st.timedeltas()
+            ),
+            typing.SupportsComplex: (
+                st.booleans()
+                | st.integers()
+                | st.floats()
+                | st.complex_numbers()
+                | st.decimals()
+                | st.fractions()
+            ),
+            typing.SupportsFloat: (
```

Comment:
also : 

i'm noticing crazy reprints using the `__or__` operator. 

```
one_of(one_of(one_of(one_of(one_of(booleans(), integers()), floats()), decimals()), fractions()), floats().map(str))
```

that's a lot of `one_of`s.... I can change to list format but i wonder if there may be another issue or i'm doing something silly.

Reply:
With this many I'd use the `one_of` strategy rather than `|` - it's exactly equivalent under the hood, but a little more readable.

Reply:
cool yeah will do